packages/next/src/client/link.tsx

```diff
@@ -161,15 +161,21 @@ function prefetch(
     prefetched.add(prefetchedKey)
   }
 
-  const prefetchPromise = isAppRouter
-    ? (router as AppRouterInstance).prefetch(href, appOptions)
```

Comment:
`router.prefetch` is sync, so this does not return a promise. looks like this part was missed when `router.prefetch` was changed from sync to async.
https://github.com/vercel/next.js/commit/f4a01b4b0e90fb1704b6a29d6dacf7863f8a87c6